### PR TITLE
HPCC-21867 HPCC Administrator Group

### PR DESCRIPTION
**Pull request by g-pan**
Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>
View in compiled PDF http://10.240.32.243/job/DocDev70-1/CE=ubuntu-14.04-amd64-1/47/artifact/build-docs/Release/docs/EN_US/ConfiguringHPCC_EN_US-7.2.15-closedown0.pdf#page=65
and see also Installing and Running and System Administrators docs. 
@JamesDeFabia please review
@RussWhitehead please review

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Successful unit testing: http://10.240.32.243/job/DocDev70-1/
 Doc Changes in Configuration manager, Installing and Running, and System Administrators guide

### CODE REVIEW

File: docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml [hunk 1]
```diff
@@ -2057,6 +2057,39 @@ sudo -u hpcc cp /etc/HPCCSystems/source/NewEnvironment.xml /etc/HPCCSystems/envi
...
         </sect3>
 
+        <sect3>
+          <title>The HPCC Administrators account</title>
+
+          <para>You can set up an HPCC administrator's account with limited
+          administrative rights. The HPCC Administrator does not have to be an
+          LDAP Administrator's account. This allows the HPCC Administrator to
+          be able to set up users, groups, and set permissions in HPCC without
+          having rights to perform other system level administrative
```
Comment (RussWhitehead) [inline docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml]: other LDAP administrative functions.

File: docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml [hunk 2]
```diff
@@ -2057,6 +2057,39 @@ sudo -u hpcc cp /etc/HPCCSystems/source/NewEnvironment.xml /etc/HPCCSystems/envi
...
+                role="bold">adminGroupName</emphasis> field. </para>
+              </listitem>
+
+              <listitem>
+                <para>Add (HPCC Administrator) users to this new group.
+                </para>
+              </listitem>
+
+              <listitem>
+                <para>Save and Deploy the new configuration file, then restart
```
Comment (RussWhitehead) [inline docs/EN_US/ConfiguringHPCC/ConfiguringHPCC.xml]: ...restart ESP...

**Review by RussWhitehead — CHANGES_REQUESTED** [review]

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
In addition to Russ Whitehead's comments, please make sure to use the apostrophe consistently.

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Inst-Mods/UserSecurityMaint.xml [hunk 3]
```diff
@@ -412,8 +412,11 @@
         <title>To promote a user to an Administrator</title>
 
         <para>To modify a users credentials you must have Administrator level
-        access. To promote a user to an HPCC Administrator, add the user to
-        the <emphasis role="bold">Administrators</emphasis> group.</para>
+        access. You can designate the HPCC Administrator account to have
+        limited permissions only relating to HPCC elements and not system
+        level administrator's rights. To promote a user to an HPCC
+        Administrator, add the user to the <emphasis
+        role="bold">Administrators</emphasis> group.</para>
```
[the remark this answers is not in the document]
Comment (g-pan) [reply docs/EN_US/Installing_and_RunningTheHPCCPlatform/Inst-Mods/UserSecurityMaint.xml]: See step 4. little further down the page.

**Comment by g-pan** [comment]
@RussWhitehead please note changes to UserSecurityMain.xml file differ from suggested
and please review
@JamesDeFabia please review

**Review by JamesDeFabia — APPROVED** [review]
Looks good to me now, but @RussWhitehead should have the last word

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Inst-Mods/UserSecurityMaint.xml [hunk 4]
```diff
@@ -464,10 +467,12 @@
                   <term>NOTE:</term>
 
                   <listitem>
-                    <para>The name of the default Administrator group could
-                    vary. For example, in Active Directory, it is
-                    "Administrators", in OpenLDAP it is "Directory
-                    Administrators".</para>
+                    <para>The name of the default Administrators group could
+                    vary. It is a configurable value defined as the the value
```
Comment (RussWhitehead) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Inst-Mods/UserSecurityMaint.xml]: the the

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Inst-Mods/hpcc_ldap.xml [hunk 5]
```diff
@@ -788,6 +788,11 @@ Ready to initialize HPCC LDAP Environment, using the following settings
        HPCC Admin User : HPCCAdmin389
 Proceed?  y/n 
 </programlisting></para>
+
+      <para>Remember that the initldap utility leverages an LDAP
```
Comment (RussWhitehead) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Inst-Mods/hpcc_ldap.xml]: initldap does not create the admin group, so you can remove these 4 lines

**Review by RussWhitehead — CHANGES_REQUESTED** [review]

**Comment by g-pan** [comment]
revised and rebased. @RussWhitehead please review

**Review by RussWhitehead — APPROVED** [review]
@g-pan Looks good to go

**Comment by JamesDeFabia** [comment]
@richardkchapman This should be good to merge now. Also please look at the Jira because Jirabot added the Pull Request but did not set it to Merge Pending.

**Pull request merged**